### PR TITLE
Implement StructureMap extraction for CHW ANC BMI questionnaire

### LINKED ISSUE #897: CHW/ANC - Extract height, weight and BMI observations and encounter using StructureMap
Origin Issue #521

Sample Observation:
Patient Weight Observation - [See sample](https://www.hl7.org/fhir/observation-example.json.html)
Patient Haight Observation - [See sample](https://www.hl7.org/fhir/observation-example-body-height.json.html)
Patient BMI Observation NB. This will have reference to the height and weight observation details together with the BMI value - [See sample](https://www.hl7.org/fhir/observation-example-bmi-using-related.json.html)

working bmi-questionnaire in the anc app is with id-1999 on local tester

CC: @f-odhiambo , @kwasim

### PR DESCRIPTION
**Pull request by ekigamba**
Fixes #897

**Type**
Choose one: Code health

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by pld** [comment]
is this still relevant? If so, can we finish it up

**Comment by ekigamba** [comment]
> is this still relevant? If so, can we finish it up

Cool

**Review by RaaziaTarique — APPROVED** [review]
LGTM

**Review by RaaziaTarique — CHANGES_REQUESTED** [review]

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/details/bmicompute/BmiQuestionnaireActivity.kt [hunk 1]
```diff
@@ -48,39 +49,41 @@ class BmiQuestionnaireActivity : QuestionnaireActivity() {
...
-      else {
-        val patientId = intent.getStringExtra(QUESTIONNAIRE_ARG_PATIENT_KEY)!!
-        showBmiDataAlert(
-          questionnaireResponse = questionnaireResponse,
-          patientId = patientId,
-          weight = inputWeight,
-          height = inputHeight,
-          computedBmi = computedBmi,
-          isUnitModeMetric = isUnitModeMetric
+      val bundle =
```
[the remark this answers is not in the document]
Comment (ekigamba) [reply android/anc/src/main/java/org/smartregister/fhircore/anc/ui/details/bmicompute/BmiQuestionnaireActivity.kt:52]: I have fixed the height observation not having the `Obs.effective`. This was missing for the standard measurement

[the remark this answers is not in the document]
Comment (ekigamba) [reply android/anc/src/main/java/org/smartregister/fhircore/anc/ui/details/bmicompute/BmiQuestionnaireActivity.kt:52]: I have also uploaded the new StructureMap

**Review by RaaziaTarique — DISMISSED** [review]

**Review by RaaziaTarique — APPROVED** [review]

**Pull request merged**